src/index.ts

```diff
@@ -107,9 +137,15 @@ const computePackageInstances = (packages: Packages, name: string, useMostCommon
     // Extract the list of unique versions for this package
     const versions:Versions = new Map();
     for (const packageInstance of packageInstances) {
-        if (versions.has(packageInstance.installedVersion)) continue;
+        if (versions.has(packageInstance.installedVersion)) {
+            const existingPackage = versions.get(packageInstance.installedVersion)!;
```

Comment:
I don't get why we need to add `isDirectDependency` to every dep in the tree.

Wouldn't checking `directDependencies.has(entryName)` when sorting the versions (line 177) suffice? Especially when we encapsulate this logic into its own strategy.

Reply:
I've moved this down here so only `Version` needs the `isDirectDependency` flag. There's no `entryName` to check in the version sorting step so it's not that easy to do it there. If we absolutely want to avoid having this, I suppose it's possible to loop over `satisfies` in the sorting function and check if any requested version is a direct dependency but that'd just be worse for performance.